### ISSUE TITLE
Bug: sam init - ScannerError

### ISSUE THREAD

**Issue by quangchustudy** [aws/aws-sam-cli#5167]
### Description:
Sam init error

### Steps to reproduce:
sam init with choosing:
Which template source would you like to use?
1 - AWS Quick Start Templates
Choose an AWS Quick Start application template
4 - Scheduled task
Which runtime would you like to use?
3 - nodejs18.x

### Observed result:

```
Which template source would you like to use?
        1 - AWS Quick Start Templates
        2 - Custom Template Location
Choice: 1

Choose an AWS Quick Start application template
        1 - Hello World Example
        2 - Multi-step workflow
        3 - Serverless API
        4 - Scheduled task
        5 - Standalone function
        6 - Data processing
        7 - Hello World Example With Powertools
        8 - Infrastructure event management
        9 - Serverless Connector Hello World Example
        10 - Multi-step workflow with Connectors
        11 - Lambda EFS example
        12 - DynamoDB Example
        13 - Machine Learning
Template: 4

Which runtime would you like to use?
        1 - dotnet6
        2 - dotnetcore3.1
        3 - nodejs18.x
        4 - nodejs16.x
        5 - nodejs14.x
        6 - nodejs12.x
Runtime: 3

Based on your selections, the only Package type available is Zip.
We will proceed to selecting the Package type as Zip.

Based on your selections, the only dependency manager available is npm.
We will proceed copying the template using npm.

Would you like to enable X-Ray tracing on the function(s) in your application?  [y/N]: y
X-Ray will incur an additional cost. View https://aws.amazon.com/xray/pricing/ for more details

Would you like to enable monitoring using CloudWatch Application Insights?
For more info, please view https://docs.aws.amazon.com/AmazonCloudWatch/latest/monitoring/cloudwatch-application-insights.html [y/N]: n

Project name [sam-app]: scheduled-task

    -----------------------
    Generating application:
    -----------------------
    Name: scheduled-task
    Runtime: nodejs18.x
    Architectures: x86_64
    Dependency Manager: npm
    Application Template: quick-start-cloudwatch-events
    Output Directory: .
    Configuration file: scheduled-task/samconfig.toml
    
    Next steps can be found in the README file at scheduled-task/README.md
        

Commands you can use next
=========================
[*] Create pipeline: cd scheduled-task && sam pipeline init --bootstrap
[*] Validate SAM template: cd scheduled-task && sam validate
[*] Test Function in the Cloud: cd scheduled-task && sam sync --stack-name {stack-name} --watch


Error: mapping values are not allowed here
  in "<unicode string>", line 44, column 14:
          Timeout: 100
                 ^
Traceback:
  File "/opt/homebrew/Cellar/aws-sam-cli/1.77.0/libexec/lib/python3.11/site-packages/click/core.py", line 1055, in main
    rv = self.invoke(ctx)
         ^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Cellar/aws-sam-cli/1.77.0/libexec/lib/python3.11/site-packages/click/core.py", line 1657, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
                           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Cellar/aws-sam-cli/1.77.0/libexec/lib/python3.11/site-packages/click/core.py", line 1404, in invoke
    return ctx.invoke(self.callback, **ctx.params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Cellar/aws-sam-cli/1.77.0/libexec/lib/python3.11/site-packages/click/core.py", line 760, in invoke
    return __callback(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Cellar/aws-sam-cli/1.77.0/libexec/lib/python3.11/site-packages/samcli/commands/init/command.py", line 132, in wrapped
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Cellar/aws-sam-cli/1.77.0/libexec/lib/python3.11/site-packages/click/decorators.py", line 84, in new_func
    return ctx.invoke(f, obj, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Cellar/aws-sam-cli/1.77.0/libexec/lib/python3.11/site-packages/click/core.py", line 760, in invoke
    return __callback(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Cellar/aws-sam-cli/1.77.0/libexec/lib/python3.11/site-packages/samcli/lib/telemetry/metric.py", line 183, in wrapped
    raise exception  # pylint: disable=raising-bad-type
    ^^^^^^^^^^^^^^^
  File "/opt/homebrew/Cellar/aws-sam-cli/1.77.0/libexec/lib/python3.11/site-packages/samcli/lib/telemetry/metric.py", line 148, in wrapped
    return_value = func(*args, **kwargs)
                   ^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Cellar/aws-sam-cli/1.77.0/libexec/lib/python3.11/site-packages/samcli/lib/utils/version_checker.py", line 42, in wrapped
    actual_result = func(*args, **kwargs)
                    ^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Cellar/aws-sam-cli/1.77.0/libexec/lib/python3.11/site-packages/samcli/cli/main.py", line 92, in wrapper
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Cellar/aws-sam-cli/1.77.0/libexec/lib/python3.11/site-packages/samcli/commands/init/command.py", line 276, in cli
    do_cli(
  File "/opt/homebrew/Cellar/aws-sam-cli/1.77.0/libexec/lib/python3.11/site-packages/samcli/commands/init/command.py", line 372, in do_cli
    do_interactive(
  File "/opt/homebrew/Cellar/aws-sam-cli/1.77.0/libexec/lib/python3.11/site-packages/samcli/commands/init/interactive_init_flow.py", line 68, in do_interactive
    generate_application(
  File "/opt/homebrew/Cellar/aws-sam-cli/1.77.0/libexec/lib/python3.11/site-packages/samcli/commands/init/interactive_init_flow.py", line 137, in generate_application
    _generate_from_use_case(
  File "/opt/homebrew/Cellar/aws-sam-cli/1.77.0/libexec/lib/python3.11/site-packages/samcli/commands/init/interactive_init_flow.py", line 284, in _generate_from_use_case
    do_generate(
  File "/opt/homebrew/Cellar/aws-sam-cli/1.77.0/libexec/lib/python3.11/site-packages/samcli/commands/init/init_generator.py", line 23, in do_generate
    generate_project(
  File "/opt/homebrew/Cellar/aws-sam-cli/1.77.0/libexec/lib/python3.11/site-packages/samcli/lib/init/__init__.py", line 131, in generate_project
    _apply_tracing(tracing, output_dir, name)
  File "/opt/homebrew/Cellar/aws-sam-cli/1.77.0/libexec/lib/python3.11/site-packages/samcli/lib/init/__init__.py", line 142, in _apply_tracing
    template_modifier.modify_template()
  File "/opt/homebrew/Cellar/aws-sam-cli/1.77.0/libexec/lib/python3.11/site-packages/samcli/lib/init/template_modifiers/cli_template_modifier.py", line 30, in modify_template
    if not self._sanity_check():
           ^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Cellar/aws-sam-cli/1.77.0/libexec/lib/python3.11/site-packages/samcli/lib/init/template_modifiers/cli_template_modifier.py", line 119, in _sanity_check
    parse_template = parse_yaml_file(self.template_location)
                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Cellar/aws-sam-cli/1.77.0/libexec/lib/python3.11/site-packages/samcli/yamlhelper.py", line 151, in parse_yaml_file
    return yaml_parse(content)
           ^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Cellar/aws-sam-cli/1.77.0/libexec/lib/python3.11/site-packages/samcli/yamlhelper.py", line 127, in yaml_parse
    return cast(Dict, yaml.safe_load(yamlstr))
                      ^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Cellar/aws-sam-cli/1.77.0/libexec/lib/python3.11/site-packages/yaml/__init__.py", line 162, in safe_load
    return load(stream, SafeLoader)
           ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Cellar/aws-sam-cli/1.77.0/libexec/lib/python3.11/site-packages/yaml/__init__.py", line 114, in load
    return loader.get_single_data()
           ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Cellar/aws-sam-cli/1.77.0/libexec/lib/python3.11/site-packages/yaml/constructor.py", line 49, in get_single_data
    node = self.get_single_node()
           ^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Cellar/aws-sam-cli/1.77.0/libexec/lib/python3.11/site-packages/yaml/composer.py", line 36, in get_single_node
    document = self.compose_document()
               ^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Cellar/aws-sam-cli/1.77.0/libexec/lib/python3.11/site-packages/yaml/composer.py", line 55, in compose_document
    node = self.compose_node(None, None)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Cellar/aws-sam-cli/1.77.0/libexec/lib/python3.11/site-packages/yaml/composer.py", line 84, in compose_node
    node = self.compose_mapping_node(anchor)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Cellar/aws-sam-cli/1.77.0/libexec/lib/python3.11/site-packages/yaml/composer.py", line 133, in compose_mapping_node
    item_value = self.compose_node(node, item_key)
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Cellar/aws-sam-cli/1.77.0/libexec/lib/python3.11/site-packages/yaml/composer.py", line 84, in compose_node
    node = self.compose_mapping_node(anchor)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Cellar/aws-sam-cli/1.77.0/libexec/lib/python3.11/site-packages/yaml/composer.py", line 133, in compose_mapping_node
    item_value = self.compose_node(node, item_key)
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Cellar/aws-sam-cli/1.77.0/libexec/lib/python3.11/site-packages/yaml/composer.py", line 84, in compose_node
    node = self.compose_mapping_node(anchor)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Cellar/aws-sam-cli/1.77.0/libexec/lib/python3.11/site-packages/yaml/composer.py", line 127, in compose_mapping_node
    while not self.check_event(MappingEndEvent):
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Cellar/aws-sam-cli/1.77.0/libexec/lib/python3.11/site-packages/yaml/parser.py", line 98, in check_event
    self.current_event = self.state()
                         ^^^^^^^^^^^^
  File "/opt/homebrew/Cellar/aws-sam-cli/1.77.0/libexec/lib/python3.11/site-packages/yaml/parser.py", line 428, in parse_block_mapping_key
    if self.check_token(KeyToken):
       ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Cellar/aws-sam-cli/1.77.0/libexec/lib/python3.11/site-packages/yaml/scanner.py", line 116, in check_token
    self.fetch_more_tokens()
  File "/opt/homebrew/Cellar/aws-sam-cli/1.77.0/libexec/lib/python3.11/site-packages/yaml/scanner.py", line 223, in fetch_more_tokens
    return self.fetch_value()
           ^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Cellar/aws-sam-cli/1.77.0/libexec/lib/python3.11/site-packages/yaml/scanner.py", line 577, in fetch_value
    raise ScannerError(None, None,

An unexpected error was encountered while executing "sam init".
Search for an existing issue:
https://github.com/aws/aws-sam-cli/issues?q=is%3Aissue+is%3Aopen+Bug%3A%20sam%20init%20-%20ScannerError
Or create a bug report:
https://github.com/aws/aws-sam-cli/issues/new?template=Bug_report.md&title=Bug%3A%20sam%20init%20-%20ScannerError
```

### Expected result:
Everything is ok

### Additional environment details (Ex: Windows, Mac, Amazon Linux etc)

1. OS: macOS Ventura13.3.1
2. `sam --version`: SAM CLI, version 1.77.0
3. AWS region: ap-southeast-1

```
# Paste the output of `sam --info` here
```

`Add --debug flag to command you are running`

**Comment by mndeveci** [comment]
Hi there,

I've tried to re-produce this issue with latest SAM CLI version, but I was able to instantiate my application and was able to run `sam build`.
```
❯ sam init

You can preselect a particular runtime or package type when using the `sam init` experience.
Call `sam init --help` to learn more.

Which template source would you like to use?
	1 - AWS Quick Start Templates
	2 - Custom Template Location
Choice: 1

Choose an AWS Quick Start application template
	1 - Hello World Example
	2 - Data processing
	3 - Hello World Example With Powertools
	4 - Multi-step workflow
	5 - Scheduled task
	6 - Standalone function
	7 - Serverless API
	8 - Infrastructure event management
	9 - Lambda Response Streaming
	10 - Serverless Connector Hello World Example
	11 - Multi-step workflow with Connectors
	12 - Lambda EFS example
	13 - DynamoDB Example
	14 - Machine Learning
Template: 5

Which runtime would you like to use?
	1 - dotnet6
	2 - nodejs18.x
	3 - nodejs16.x
	4 - nodejs14.x
	5 - nodejs12.x
Runtime: 2

Based on your selections, the only Package type available is Zip.
We will proceed to selecting the Package type as Zip.

Based on your selections, the only dependency manager available is npm.
We will proceed copying the template using npm.

Would you like to enable X-Ray tracing on the function(s) in your application?  [y/N]: y
X-Ray will incur an additional cost. View https://aws.amazon.com/xray/pricing/ for more details

Would you like to enable monitoring using CloudWatch Application Insights?
For more info, please view https://docs.aws.amazon.com/AmazonCloudWatch/latest/monitoring/cloudwatch-application-insights.html [y/N]:

Project name [sam-app]:

    -----------------------
    Generating application:
    -----------------------
    Name: sam-app
    Runtime: nodejs18.x
    Architectures: x86_64
    Dependency Manager: npm
    Application Template: quick-start-cloudwatch-events
    Output Directory: .
    Configuration file: sam-app/samconfig.toml

    Next steps can be found in the README file at sam-app/README.md


Commands you can use next
=========================
[*] Create pipeline: cd sam-app && sam pipeline init --bootstrap
[*] Validate SAM template: cd sam-app && sam validate
[*] Test Function in the Cloud: cd sam-app && sam sync --stack-name {stack-name} --watch
❯ cd sam-app
❯ sam build
Starting Build use cache
Manifest file is changed (new hash: 57574ed173cdc6a98b283bd7b00b44ca) or dependency folder
(.aws-sam/deps/63595733-3d68-4908-93a6-21ad4afbf962) is missing for (ScheduledEventLogger), downloading dependencies and
copying/building source
Building codeuri: /private/tmp/sam-init-error/sam-app runtime: nodejs18.x metadata: {} architecture: x86_64 functions:
ScheduledEventLogger
Running NodejsNpmBuilder:NpmPack
Running NodejsNpmBuilder:CopyNpmrcAndLockfile
Running NodejsNpmBuilder:CopySource
Running NodejsNpmBuilder:NpmInstall
Running NodejsNpmBuilder:CleanUp
Running NodejsNpmBuilder:CopyDependencies
Running NodejsNpmBuilder:CleanUpNpmrc
Running NodejsNpmBuilder:LockfileCleanUp
Running NodejsNpmBuilder:LockfileCleanUp
```

I saw that you are running an old version of SAM CLI, can you try to update your installation to see if the problem goes away?

**Comment by hawflau** [comment]
Hey @quangchustudy the error you saw was due to bug that we fixed in [v1.79.0](https://github.com/aws/aws-sam-cli/releases/tag/v1.79.0). Please update to latest version.

Closing. Feel free to reopen if you still see any problem.
